anyio/__init__.py

```diff
@@ -276,11 +276,18 @@ def aopen(file: Union[str, 'os.PathLike', int], mode: str = 'r', buffering: int
 async def connect_tcp(
     address: IPAddressType, port: int, *, ssl_context: Optional[SSLContext] = None,
     autostart_tls: bool = False, bind_host: Optional[IPAddressType] = None,
-    bind_port: Optional[int] = None, tls_standard_compatible: bool = True
+    bind_port: Optional[int] = None, tls_standard_compatible: bool = True,
+    happy_eyeballs_delay: float = 0.25
 ) -> SocketStream:
     """
     Connect to a host using the TCP protocol.
 
+    This function implements the stateless version of the Happy Eyeballs algorithm (RFC 6555).
+    If ``address`` is a host name that resolves to multiple IP addresses, each one is tried until
+    one connection attempt succeeds. If the first attempt does not connected within 300
+    milliseconds, a second attempt is started using the next address in the list, and so on.
```

Comment:
250 … 300 … ;-)

Reply:
Right...I changed this later.

Reply:
I was not aware that there is an updated version of the Happy Eyeballs RFC.

Reply:
Trio does not do this; I wonder if there's any merit in AnyIO adding this.

Reply:
Trio probably should also. I'll look into it.